### ISSUE TITLE
Dynamic import with destructuring false positive on unused export

### ISSUE THREAD

**Issue by shawnmcknight** [webpro-nl/knip#688]
I've created a simple reproduction at https://github.com/shawnmcknight/knip-dynamic-import-destructuring.

Dynamic imports of the signature `import("./components.ts").then(({ Apple }) => Apple);` are reported as unused exports while dynamic imports of the signature `import("./components.ts").then((mod) => mod.Orange);` are not reported as unused exports.

As illustrated in the reproduction, I have a `components.ts` file that is:
```ts
export const Apple = () => "Apple";
export const Orange = () => "Orange";
```

and an `index.ts` file that is:
```ts
import("./components.ts").then(({ Apple }) => Apple);
import("./components.ts").then((mod) => mod.Orange);
```

When running knip the output is:
```
Unused exports (1)                
Apple  unknown  components.ts:1:14
```

Please let me know if there is anything else I can provide to assist!

**Comment by webpro** [comment]
Thanks @shawnmcknight, great report + repro 🙏

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.22.1. See [Release 5.22.1](https://github.com/webpro-nl/knip/releases/tag/5.22.1) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._
